src/traits.rs

```diff
@@ -99,20 +110,84 @@ impl Lerp for u32 {
     type Ratio = f64;
 }
 
+impl Lerp for f32 {
+    type Ratio = f32;
+
+    fn lerp(a: Self, b: Self, ratio: Self::Ratio) -> Self {
+        a + (b - a) * ratio
+    }
+}
+
+
+pub trait Sample: Primitive + 'static {
```

Comment:
Opening this to deliberate on the name of the trait (whether or not we merge it with the current `Primitive`). My problem with the name `Sample` is that in GPU is refers to one computed color instance of a pixel in a fragment—of which there could be multiple in multisampled graphics pipelines.

Another alternative would be `Channel` but this one has the added meaning of an _interpretation_ of the number. For example, red channel, green channel of an RGB-ish pixel make sense but 'the u8 channel' doesn't.

The last name that comes to mind is `Component` which, at least in my intuitive interpretation, is very closely related to the bit-encoding. That could be fitting here. We currently only provide color representations with homogeneous component depths. So the description of this trait could be summed up similar to:

> A trivial type that can be used for any channel value in a color representation.

Reply:
Agreed :)

I'd favor `Component`, but it is an overused word, maybe `PixelComponent`? Or, instead of `Channel`, we could use `ChannelType` or `ChannelPrimitive`?

Reply:
I think it would be nice to also rename `trait Pixel { type Subpixel: PixelComponent; }` to `.. { type Component: PixelComponent }`. What do you think?

Reply:
Should we rename the `sample` in all places in the project, for example also  `Buffer::into_flat_samples` to  `Buffer::into_flat_components`?

Reply:
> I think it would be nice to also rename trait Pixel { type Subpixel: PixelComponent; ... } to trait Pixel { type Component: PixelComponent ... }. What do you think?

> Should we rename the sample in all places in the project, for example also Buffer::into_flat_samples to Buffer::into_flat_components?

Maybe but it's examples of those 'nice but high churn from breakage' changes. For the methods I would deprecate/hide the existing implementations and then add the new naming on top (and remove in 0.25). I wouldn't rename the associated type as there is no compatibility/deprecation on it at the moment. The trait rename is acceptable in so far as the trait itself will undergo a major change.

Reply:
Also, please only do that sort of widespread change in a standalone PR! If you are purely changing names it can just be a matter of skimming everything, but if there are functional changes as well the reviewing is much more time consuming

Reply:
For a merged PixelComponent trait, I would also suggest making it sealed so that we can guarantee there are no downstream implementations.

Reply:
Alright, I'll rename only the types that were introduced in this pr and make the trait sealed :)

Reply:
To be frank I think sealing the trait in this one makes it _worse_. We don't gain any additional ability to change the associated constants or to resolve the color space tension. We can't rely on the sealing for any effect as most of our implementations are generic over `<T: Primitive>` anyways.

But now no one downstream can change our defaults by, e.g. implementing a `Component` that has its maximum value at a different constant or one for their own half-floats, for non-NaN floats etc.